src/virtualenv/seed/embed/via_app_data/pip_install/base.py

```diff
@@ -31,6 +31,16 @@ def _sync(self, src, dst):
 
     def install(self, version_info):
         self._extracted = True
+        # cleanup existing version's .dist-info if exists
```

Comment:
Hmmm, any reason we can't just tell users to use ``--clean`` in this case?

Needs test though :+1:

Reply:
Because this is a case where we're merely overlaying without erasing.

Reply:
I'm not sure that's a supported use case though 😬 there's no guarantee that the outcome would always be working... E.g. you overlay a python 3.10 over a python 3.9, the c extension packages now can throw runtime error 😬

Reply:
There are 3 cases:
1. Install into a non-existent directory
2. Install into an existent directory
3. Install into an existent directory with "--clean" option.

I'm happy with [2], as it speeds things up, but it needs to work correctly, hence this PR. If the desire is to only keep [1] and [3] then installing into an existent directory should fail outright. Doing so will probably create a ton of regression for people who use it for overlay functionality.

I'll add the tests.

Reply:
> E.g. you overlay a python 3.10 over a python 3.9, the c extension packages now can throw runtime error grimacing

Sure. But that's up to the user to deal with that issue. We use it as part of the automation and the specific python version is guaranteed. Cleaning will require pip installing dependencies every time and that will slow things down dramatically.